be/src/olap/rowset/segment_v2/column_writer.cpp

```diff
@@ -274,6 +274,7 @@ Status ScalarColumnWriter::init() {
     // create page builder
     PageBuilderOptions opts;
     opts.data_page_size = _opts.data_page_size;
+    opts.dict_page_size = _opts.data_page_size; // init smaller dict page, grow if need more
```

Comment:
We can leave `dict_page_size` as 1M, and not to change the `data_page_size` when we initialize the `_dict_builder`. So you can reserve the faststring with `min(data_page_size, dict_page_size)` and use `dict_page_size` to check if the page is full.

Reply:
Thanks for your suggestions. The code is modified accordingly and tested again using the config above. Everything goes okay now I hope.